src/main/webapp/app/exercises/file-upload/participate/file-upload-submission.service.ts

```diff
@@ -36,11 +36,16 @@ export class FileUploadSubmissionService {
     /**
      * Returns File Upload submission from the server
      * @param fileUploadSubmissionId the id of the File Upload submission
+     * @param correctionRound
+     * @param resultId
      */
-    get(fileUploadSubmissionId: number, correctionRound = 0): Observable<HttpResponse<FileUploadSubmission>> {
+    get(fileUploadSubmissionId: number, correctionRound = 0, resultId?: number): Observable<HttpResponse<FileUploadSubmission>> {
         const url = `api/file-upload-submissions/${fileUploadSubmissionId}`;
         let params = new HttpParams();
-        if (correctionRound !== 0) {
+        if (resultId && resultId > 0) {
+            // in case resultId is set, we do not need the correction round
+            params = params.set('resultId', resultId!.toString());
+        } else {
             params = params.set('correction-round', correctionRound.toString());
```

Comment:
Is there a reason we should not send the correction-round, if we send the result Id? I get that we theoretically don't need it, but I don't know for sure how the server handles this. Maybe there could be a conflict when the correctionRound in the client is 1, we only send the resultId, and the server, by default assumes the correctionRound is 0.

Reply:
In the case where we have the resultId we don't know the correction round, that's why we use the resultId instead.